### PR TITLE
[CELEBORN-1195] Use batch rack resolve when restore meta from file

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?、
```java
org.apache.celeborn.service.deploy.master.clustermeta.AbstractMetaManager#restoreMetaFromFile
```

### Why are the changes needed?
When the number of workers is large, the performance of parsing one by one will decrease.

YARN-9332. RackResolver tool should accept multiple hosts
https://issues.apache.org/jira/browse/YARN-9332

### CODE REVIEW

File: master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java [hunk 1]
```diff
@@ -281,25 +283,28 @@ public void restoreMetaFromFile(File file) throws IOException {
...
+              .toSeq();
+      scala.collection.immutable.Map<String, Node> resolveMap =
+          rackResolver.resolveToMap(workerHostSeq);
+      workers.addAll(
+          workerInfoSet.stream()
+              .peek(
                   workerInfo -> {
                     // Reset worker's network location with current master's configuration.
                     workerInfo.networkLocation_$eq(
-                        rackResolver.resolve(workerInfo.host()).getNetworkLocation());
```
Comment (cxzl25) [inline master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java:293]: When the number of workers is large, the performance of parsing one by one will decrease.

YARN-9332. RackResolver tool should accept multiple hosts
https://issues.apache.org/jira/browse/YARN-9332

**Review by onebox-li — APPROVED** [review]

**Comment by FMX** [comment]
Although Celeborn is using Hadoop 3.3.6 as default. But Celeborn supports Hadoop 2.x, I think this PR will cause Celeborn can not compile successfully when using Hadoop 2.x.
I think you can verify this point.

**Comment by cxzl25** [comment]
> But Celeborn supports Hadoop 2.x, I think this PR will cause Celeborn can not compile successfully when using Hadoop 2.x.

Because we use the `FileSystem.listStatusIterator` API in `Master.scala`, its minimum version is hadoop-client 2.7.0
If we use hadoop-client 2.6.0 we will get the error as below.
```bash
./build/make-distribution.sh -Pspark-3.1 -Dhadoop.version=2.6.0 
```
```
incubator-celeborn/master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala:810: value listStatusIterator is not a member of org.apache.hadoop.fs.FileSystem
```
HADOOP-10987. Provide an iterator-based listing API for FileSystem
https://issues.apache.org/jira/browse/HADOOP-10987
Fix Version/s: 2.7.0

```bash
./build/make-distribution.sh -Pspark-3.1 -Dhadoop.version=2.7.0
```
```
build success
```


`DNSToSwitchMapping` has supported list parameters in very early hadoop versions, but `org.apache.hadoop.yarn.util.RackResolver` does not support list parameters. 
It is only supported in YARN-9332, so we now use CelebornRackResolver, refer to The implementation of SparkRackResolver bypasses YARN's `RackResolver` and directly calls `DNSToSwitchMapping`, so there will be no problem of breaking the compilation.

**Comment by FMX** [comment]
Thanks for your explanation. This PR LGTM. Merged into main(v0.5.0).

**Review by FMX — APPROVED** [review]
LGTM.

**Pull request closed without merging**